### PR TITLE
Issue #12111: upgrades maven version for contributors

### PR DESCRIPTION
**Pull request by rnveach**
Issue #12111

### CODE REVIEW

File: src/xdocs/beginning_development.xml [hunk 1]
```diff
@@ -17,7 +17,7 @@
     </section>
     <section name="Before development">
       <p>
-        1. Ensure that Git, Java JDK >= 11 until JDK 17, maven >= 3.0.1 are installed.<br/>
+        1. Ensure that Git, Java JDK >= 11 until JDK 17, maven >= 3.3.9 are installed.<br/>
         You can find information about development environment preparation here:
```
Comment (romani) [inline src/xdocs/beginning_development.xml:21]: We usually need to update pom.xml and CIs.

Comment (rnveach) [reply src/xdocs/beginning_development.xml:21]: Updated one entry in the POM. CIs would have been failing if they used a version earlier than this.

Regardless, I couldn't find anything else to update. Let me know if there is anything else.

Comment (romani) [reply src/xdocs/beginning_development.xml:21]: if CI pass, ok to merge

**Comment by rnveach** [comment]
@nick-mancuso @Vyom-Yadav Not related to my changes. Restarting.
````
New surviving mutation(s) found:

Source File: "Checker.java"
Class: "com.puppycrawl.tools.checkstyle.Checker"
Method: "finishLocalSetup"
Line Contents: "Violation.setLocale(locale);"
Mutator: "org.pitest.mutationtest.engine.gregor.mutators.VoidMethodCallMutator"
Description: "removed call to com/puppycrawl/tools/checkstyle/api/Violation::setLocale"
Line Number: 434
Error: Process completed with exit code 1.
```

**Review by romani — APPROVED** [review]
Ok to merge

**Review by strkkk — APPROVED** [review]

**Pull request merged**